### PR TITLE
[docs] add note about upgrading to 1.8.0 in openshift

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

Adds note in our openshift doc about a known issue when upgrading to 1.8.0 in openshift

### Motivation

https://github.com/DataDog/datadog-operator/issues/1389

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### Checklist

- [X] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [X] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: docs/install-openshift.md [hunk 1]
```diff
@@ -120,6 +120,33 @@ This issue is resolved in Datadog Operator bundle 1.4.2+. Use the OperatorHub UI
...
+
+For more details, see [Removing an old version][8] from the Kubernetes Custom Resources documentation. If this command was not run before the upgrade and you encountered the error, please follow the **Resolution** steps below.
+
+#### Resolution
+
+1. Remove `v1alpha1` from DatadogAgent CRD: `oc patch customresourcedefinitions datadogagents.datadoghq.com --subresource='status' --type='merge' -p '{"status":{"storedVersions":["v2alpha1"]}}'`
+2. Remove both `ClusterServiceVersion` (current and failed upgrade): `oc delete csv datadog-operator.v1.7.0 datadog-operator.v1.8.0`
+3. Remove the status from the `Subscription` to force a new reconciliation: `oc patch sub datadog-operator-certified --subresource='status' --type='json' -p='[{"op": "remove", "path": "/status"}]'`
+4. The reconciliation should lead to a new `ClusterServiceVersion` (identical to the one preceeding the upgrade). Depending on the `installPlanApproval`, you might have to approve the `InstallPlan`.
+5. Proceed with the upgrade normally from the OperatorHub.
```
Comment (celenechang) [inline docs/install-openshift.md:146]:
```suggestion
1. Remove `v1alpha1` from the DatadogAgent CRD: `oc patch customresourcedefinitions datadogagents.datadoghq.com --subresource='status' --type='merge' -p '{"status":{"storedVersions":["v2alpha1"]}}'`
2. Remove both `ClusterServiceVersion` resources (current and failed upgrade): `oc delete csv datadog-operator.v1.7.0 datadog-operator.v1.8.0`
3. Remove the status from the `Subscription` to force a new reconciliation: `oc patch sub <subscription name> --subresource='status' --type='json' -p='[{"op": "remove", "path": "/status"}]'`
4. A new `ClusterServiceVersion` should be created (identical to the one preceding the upgrade). Depending on the `installPlanApproval` configuration, you may need to approve the `InstallPlan`.
5. Proceed with the upgrade as usual from OperatorHub.
```

**Review by celenechang — COMMENTED** [review]
thanks for the updates! could we restructure it a bit so that it's clearer there are two possible courses of action - one _before_ upgrade and one _after_ the error is viewed?

perhaps a subheader for each, and the bulk of the explanation in the _after_ section

File: docs/install-openshift.md [hunk 2]
```diff
@@ -120,6 +120,43 @@ This issue is resolved in Datadog Operator bundle 1.4.2+. Use the OperatorHub UI
...
+
+Datadog Operator bundle 1.8.0 includes the removal of the deprecated `v1alpha1` CRD version. If an older bundle installation stored this version in the DatadogAgent CRD `status` field, the Operator Lifecycle Manager automatically prevents the upgrade from completing. For more details, see [Removing an old version][8] from the Kubernetes Custom Resources documentation.
+
+#### Identification and remediation
+You can identify whether or not you are impacted by running:
+```bash
+oc get customresourcedefinitions datadogagents.datadoghq.com -osjon | jq .status.storedVersions
+```
+
+* If the returned array contains the entry `v1alpha1`, you are impacted and the rest of this guide should be followed. There are **two different courses of actions**, depending on whether or not the upgrade was attempted.
```
Comment (celenechang) [inline docs/install-openshift.md:142]:
```suggestion
* If the returned array contains the entry `v1alpha1`, you are impacted and the rest of this guide should be followed. There are **two different courses of action**, depending on whether or not the upgrade was attempted.
```

File: docs/install-openshift.md [hunk 3]
```diff
@@ -120,6 +120,43 @@ This issue is resolved in Datadog Operator bundle 1.4.2+. Use the OperatorHub UI
...
+* If not, ignore this guide and proceed with the installation/upgrade as usual.
+
+**Note**: During these steps, the Datadog Agents will keep running, ensuring your observability is not disrupted.
+
+##### Situation 1: Identification before upgrading to 1.8.0
```
Comment (celenechang) [inline docs/install-openshift.md:147]:
```suggestion
##### Scenario 1: Presence of v1alpha1 CRD confirmed before upgrading to 1.8.0
```

File: docs/install-openshift.md [hunk 4]
```diff
@@ -120,6 +120,43 @@ This issue is resolved in Datadog Operator bundle 1.4.2+. Use the OperatorHub UI
...
+
+1. Remove `v1alpha1` from the DatadogAgent CRD: `oc patch customresourcedefinitions datadogagents.datadoghq.com --subresource='status' --type='merge' -p '{"status":{"storedVersions":["v2alpha1"]}}'`
+2. Proceed with the upgrade as usual from the OperatorHub UI.
+
+##### Situation 2: Identification after attempting to upgrade to 1.8.0
```
Comment (celenechang) [inline docs/install-openshift.md:152]:
```suggestion
##### Scenario 2: Presence of v1alpha1 CRD confirmed after attempting to upgrade to 1.8.0
```

File: docs/install-openshift.md [hunk 5]
```diff
@@ -120,6 +120,43 @@ This issue is resolved in Datadog Operator bundle 1.4.2+. Use the OperatorHub UI
...
+
+Datadog Operator bundle 1.8.0 includes the removal of the deprecated `v1alpha1` CRD version. If an older bundle installation stored this version in the DatadogAgent CRD `status` field, the Operator Lifecycle Manager automatically prevents the upgrade from completing. For more details, see [Removing an old version][8] from the Kubernetes Custom Resources documentation.
+
+#### Identification and remediation
+You can identify whether or not you are impacted by running:
+```bash
+oc get customresourcedefinitions datadogagents.datadoghq.com -osjon | jq .status.storedVersions
+```
+
+* If the returned array contains the entry `v1alpha1`, you are impacted and the rest of this guide should be followed. There are **two different courses of action**, depending on whether or not the upgrade was attempted.
```
Comment (alai97) [inline docs/install-openshift.md:142]:
```suggestion
* If the returned array contains the entry `v1alpha1`, you are impacted and should follow the rest of this guide. There are **two different courses of action**, depending on whether or not the upgrade was attempted.
```

**Review by alai97 — APPROVED** [review]
Looks good, one suggestion.

**Review by tbavelier — APPROVED** [review]
LGTM

**Pull request merged**